<?xml version="1.0" ?>
<launch>
    <param name="urdfFile" value="$(find ocs2_pros_gazebo)/urdf/pros.urdf"/>
    <param name="taskFile" value="$(find ocs2_pros_controllers)/config/task.info"/>
    <param name="libFolder"  value="$(find ocs2_pros_interface)/auto_generated/pros"/>
    <rosparam file="$(find ocs2_pros_controllers)/config/controllers.yaml" command="load"/>
    
    
    <!-- controller manager加载joint_state_controller 和 legged_controller(只加载名称不启动) -->
    <!-- <node name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/pros_controller"/> -->
    
    <!-- 将gaitCommandFile加载到KeyboardPublisher中并循环监听模式切换  -->
    <!-- <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command" output="screen"/> -->
    <include file="$(find ocs2_pros_controllers)/launch/visualize.launch">
        <arg name="urdfFile" value="$(find ocs2_pros_gazebo)/urdf/pros.urdf" />
    </include>
    <node pkg ="ocs2_pros_controllers" type="pros_mpc_node"  name="pros_mpc_node"
        output="screen" launch-prefix="" />
    
    <node pkg="ocs2_pros_controllers" type="pros_dummy_mrt_node" name="pros_dummy_mrt_node" launch-prefix="gnome-terminal --" />
    <node pkg="ocs2_pros_controllers" type="pros_target" name="pros_target" output="screen" launch-prefix="" />

    <param name="robot_" value="urdf"/>

</launch>
